File: services/api-fastapi/app/permissions.py
"""Canonical list of individually-grantable dashboard tabs (profiles.extra_pages).

Mirrored exactly in apps/dashboard-web/client/src/lib/permissions.ts -- keep
both lists in sync when adding/removing a grantable tab. Keys match the
dashboard's route paths 1:1 so they read the same in the UI checklist, the
API payload, and require_role_or_grant() call sites.

Four keys (command-center, trends, pnl, oishii-ai) are normally
executive-only tabs -- granting one of these to a non-executive is a bigger
step up than granting a manager-tier tab, so app/routers/hr.py restricts who
may hand those four out (see _EXECUTIVE_ONLY_GRANTS there).
"""

GRANTABLE_PAGES: dict[str, str] = {
    "pending-orders": "Table Orders",
    "online-orders": "Online Orders",
    "stock": "Stock & Inventory (manage)",
    "business-day-report": "Business Day Report",
    "refund-approval": "Refund Approval",
    "reviews": "Customer Reviews",
    "loss-log": "Loss Log",
    "utility-log": "Utility Log",
    "pos-management": "POS Management",
    "employees": "Employees",
    "hr-attendance": "HR Attendance",
    "hr-payroll": "Payroll",
    "hr-holiday-calendar": "Holiday Calendar",
    "hr-payroll-settings": "Payroll Settings",
    "command-center": "Command Center",
    "trends": "Trend Analysis",
    "pnl": "P&L",
    "oishii-ai": "Oishii AI",
}

EXECUTIVE_ONLY_GRANTS: frozenset[str] = frozenset({"command-center", "trends", "pnl", "oishii-ai"})

File: apps/dashboard-web/client/src/lib/permissions.ts
/**
 * Canonical list of individually-grantable dashboard tabs (profiles.extra_pages).
 *
 * Mirrored exactly in services/api-fastapi/app/permissions.py -- keep both
 * lists in sync when adding/removing a grantable tab. Keys match the
 * dashboard's route paths 1:1.
 *
 * Four keys (command-center, trends, pnl, oishii-ai) are normally
 * executive-only tabs -- the backend restricts who may grant those four
 * (see app/permissions.py's EXECUTIVE_ONLY_GRANTS); this list mirrors that
 * so the checklist UI can grey them out for a non-executive caller too.
 */

export interface GrantablePage {
  key: string;
  label: string;
  group: string;
}

export const GRANTABLE_PAGES: GrantablePage[] = [
  { key: 'pending-orders', label: 'Table Orders', group: 'POS & Orders' },
  { key: 'online-orders', label: 'Online Orders', group: 'POS & Orders' },
  { key: 'stock', label: 'Stock & Inventory (manage)', group: 'Stock & Inventory' },
  { key: 'business-day-report', label: 'Business Day Report', group: 'Business & Finance' },
  { key: 'refund-approval', label: 'Refund Approval', group: 'Business & Finance' },
  { key: 'reviews', label: 'Customer Reviews', group: 'Business & Finance' },
  { key: 'loss-log', label: 'Loss Log', group: 'Business & Finance' },
  { key: 'utility-log', label: 'Utility Log', group: 'Business & Finance' },
  { key: 'pos-management', label: 'POS Management', group: 'Business & Finance' },
  { key: 'employees', label: 'Employees', group: 'HR' },
  { key: 'hr-attendance', label: 'HR Attendance', group: 'HR' },
  { key: 'hr-payroll', label: 'Payroll', group: 'HR' },
  { key: 'hr-holiday-calendar', label: 'Holiday Calendar', group: 'HR' },
  { key: 'hr-payroll-settings', label: 'Payroll Settings', group: 'HR' },
  { key: 'command-center', label: 'Command Center', group: 'Executive Analytics' },
  { key: 'trends', label: 'Trend Analysis', group: 'Executive Analytics' },
  { key: 'pnl', label: 'P&L', group: 'Executive Analytics' },
  { key: 'oishii-ai', label: 'Oishii AI', group: 'Executive Analytics' },
];

export const EXECUTIVE_ONLY_GRANTS = new Set(['command-center', 'trends', 'pnl', 'oishii-ai']);
